extension/js/common/api/key-server/wkd.ts

```diff
@@ -36,24 +40,41 @@ export class Wkd extends Api {
       // the proper fix would be to run encodeZBase32 through background scripts
       return { pubkey: null, pgpClient: null };
     }
-    const hu = opgp.util.encodeZBase32(await opgp.crypto.hash.digest(opgp.enums.hash.sha1, Buf.fromUtfStr(user)));
-    // todo - could also search on `https://openpgpkey.{domain}/.well-known/openpgpkey/{domain}/hu/{hu}?l={user}`
-    const url = `https://${recipientDomain}/.well-known/openpgpkey/hu/${hu}?l=${encodeURIComponent(user)}`;
+    const directDomain = recipientDomain.toLowerCase();
+    const advancedDomainPrefix = (directDomain === 'localhost') ? '' : 'openpgpkey.';
+    const hu = opgp.util.encodeZBase32(await opgp.crypto.hash.digest(opgp.enums.hash.sha1, Buf.fromUtfStr(user.toLowerCase())));
+    const directHost = (typeof this.port === 'undefined') ? directDomain : `${directDomain}:${this.port}`;
+    const advancedHost = `${advancedDomainPrefix}${directHost}`;
+    const userPart = `hu/${hu}?l=${encodeURIComponent(user)}`;
+    const advancedUrl = `${this.protocol}://${advancedHost}/.well-known/openpgpkey/${directDomain}`;
+    const directUrl = `${this.protocol}://${directHost}/.well-known/openpgpkey`;
     let binary: Buf;
-    try {
-      binary = await Wkd.download(url, undefined, 4);
-    } catch (e) {
-      if (ApiErr.isNotFound(e) || ApiErr.isNetErr(e)) {
+    let validUrl: string;
+    for (const url of [advancedUrl, directUrl]) {
+      try {
+        await Wkd.download(`${url}/policy`, undefined, 4);
+        console.info(`Policy found: ${url}/policy`);
+        validUrl = url;
+        break;
+      } catch (e) {
+        if (ApiErr.isNotFound(e) || ApiErr.isNetErr(e)) {
+          continue;
+        }
+        Catch.report(`Wkd.lookupEmail error retrieving policy file ${url}/policy: ${String(e)}`);
```

Comment:
I'd ignore this too - if the policy is returning a 500, it's probably because they don't have WKD set up in the first place and it's some other backend responding / getting confused.